src/aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -793,10 +794,9 @@ void AWSClient::AddChecksumToRequest(const std::shared_ptr<Aws::Http::HttpReques
         {
             if (request.IsStreaming() && checksumValueAndAlgorithmProvided)
             {
-                auto hash = Aws::MakeShared<Crypto::CRC32>(AWS_CLIENT_LOG_TAG);
-                auto precalculatedValue = request.GetHeaders().find(checksumType)->second;
-                hash->SetPrecalculatedHash(precalculatedValue);
-                httpRequest->SetRequestHash(checksumAlgorithmName, hash); }
+                auto hash = Aws::MakeShared<Crypto::PrecalculatedHash>(AWS_CLIENT_LOG_TAG,request.GetHeaders().find(checksumType)->second);
```

Comment:
we could likely factor all of this out into a single `if (request.IsStreaming() && checksumValueAndAlgorithmProvided)` statement instead of creating one for each checksum algo